### PR TITLE
FRW-1442 Migration guide for Country, Currencty, Locale modules

### PR DESCRIPTION
**Pull request by geega**
Ticket: https://spryker.atlassian.net/browse/FRW-1442

### CODE REVIEW

File: _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+This document describes how to upgrade the Country module.
+
+## Upgrading from version 3.* to version 4.0.0
+
+In this new version of the `Country` module, we have added support configuration currency for each store in database.
+With the `Country` module version 4 we have added the `spy_country_store` database table to persist stores-countries in ZED.
```
Comment (dereuromark) [inline _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md:6]: ZED is not an acronym
Zed (!)

File: _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+```
+
+2. Install the `CountryDataImport` module for import data for populate configuration tables in database.
+
+```bash
+composer require spryker/country-data-import
+```
+3. Run `vendor/bin/console transfer:generate` to update the transfer objects.
+
+4. Run `vendor/bin/console propel:migrate` to apply the database changes.
```
Comment (asmarovydlo) [inline _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md:27]: We need to run `propel:install` instead, to generate entities. The migration does not do it.
Applicable to all similar places in the PR.

File: _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+5. Add new plugins to dependency providers:
+
+**Integrate console commands**
+
+
+```php
+<?php
+
+namespace Pyz\Zed\Console;
```
Comment (asmarovydlo) [inline _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md:37]: File name is missing for this codesample, please check all similar cases in the PR.

File: _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md [hunk 4]
```diff
@@ -1,3 +1,272 @@
...
+            ...        
+        ];
+    }
+    
+    /**
+     * @param \Spryker\Zed\Kernel\Container $container
+     *
+     * @return array<\Spryker\Shared\ApplicationExtension\Dependency\Plugin\ApplicationPluginInterface>
+     */
+    public function getApplicationPlugins(Container $container): array
```
Comment (asmarovydlo) [inline _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md:106]: `Container` should be added to `use` section.

File: _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md [hunk 5]
```diff
@@ -1,3 +1,272 @@
...
+```php
+<?php
+
+namespace Pyz\Zed\Console;
+
+use Spryker\Zed\DataImport\Communication\Console\DataImportConsole;
+use Spryker\Zed\Console\ConsoleDependencyProvider as SprykerConsoleDependencyProvider;
+use Spryker\Zed\Locale\Communication\Plugin\Console\LocaleApplicationPlugin;
+use Spryker\Zed\LocaleDataImport\LocaleDataImportConfig;
+use Spryker\Zed\Locale\Communication\Plugin\Console\LocaleApplicationPlugin;
```
Comment (asmarovydlo) [inline _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md:77]: Locale application plugin is duplicated.

File: _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+{
+    ... 
+    /**
+     * @return array<\Spryker\Zed\DataImport\Dependency\Plugin\DataImportPluginInterface>
+     */
+    protected function getDataImporterPlugins(): array
+    {
+        return [
+            ...
+            new CountryStoreDataImportPlugin(),
```
Comment (asmarovydlo) [inline _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md:88]: Integration of the dynamic store will be part of `Dynamic Store Integration` guide.
**Please keep in those Migration Guides only needed migration steps to update modules to the new major** (so functionality that customers have in their projects works with the new version), we do not need to configure functionality that is not related to the minimal necessary steps to update the module.
**Applicable to all 4 IGs in the PR.**

**Review by asmarovydlo — DISMISSED** [review]

File: _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md (hunk 6 above)
Comment (geega) [reply _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md]: Yes. If only update this modules don't need integrate data import and plugins, I checked it via replace this modules + dependency

File: _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+*Estimated migration time: 5 min*
+
+To upgrade to the new version of the module, do the following:
+
+1. Upgrade the `Country` module to the new version:
+
+```bash
+composer require spryker/country:"^4.0.0" --update-with-dependencies
+```
+
```
Comment (asmarovydlo) [inline _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md:19]: https://github.com/spryker/spryker/pull/9457/files?file-filters%5B%5D=.csv&file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md&file-filters%5B%5D=.neon&file-filters%5B%5D=.php&file-filters%5B%5D=.twig&file-filters%5B%5D=.txt&file-filters%5B%5D=.xml&file-filters%5B%5D=.yaml&file-filters%5B%5D=.yml&file-filters%5B%5D=No+extension&file-filters%5B%5D=dotfile&show-viewed-files=true&show-deleted-files=false#diff-0b92f94ce075a7d30216f603b5ef8348e9749ad6f66f7da2d9b6190189f947df

This change is not mentioned but requires some action items.

Comment (geega) [reply _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md:19]: I can't open this link

File: _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
+This document describes how to upgrade the Country module.
+
+## Upgrading from version 3.* to version 4.0.0
+
+In this new version of the `Country` module, we have added support configuration currency for each store in database.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md:5]:
```suggestion
In this new 4.0.0 version of the `Country` module, we have added support configuration currency for each store in the database.
```

File: _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+With the `Country` module version 4 we have added the `spy_country_store` database table to persist stores-countries in Zed.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md:6]:
```suggestion
With the new 4.0.0 version of the `Country` module, we have added the `spy_country_store` database table to persist stores-countries in Zed.
```

File: _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+You can find more details about the changes on the [Country module](https://github.com/spryker/country/releases) release page.
+
+*Estimated migration time: 5 min*
```
Comment (AlexSlawinski) [inline _includes/pbc/all/upgrade-modules/upgrade-the-country-module.md:10]:
```suggestion
*Estimated migration time: 5 minutes*
```

File: _includes/pbc/all/upgrade-modules/upgrade-the-currency-module.md [hunk 11]
```diff
@@ -1,4 +1,29 @@
+This document describes how to upgrade the Currency module.
 
+## Upgrading from version 3.* to version 4.0.0
+
+In this new version of the `Currency` module, we have added support configuration currency for each store in database.
+With the `Currency` module version 4 we have added the `spy_currency_store` database table to persist stores-locales in Zed.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/upgrade-modules/upgrade-the-currency-module.md:6]:
```suggestion
With the new 4.0.0 version of the `Currency` module, we have added the `spy_currency_store` database table to persist stores-locales in Zed.
```

File: _includes/pbc/all/upgrade-modules/upgrade-the-currency-module.md [hunk 12]
```diff
@@ -1,4 +1,29 @@
...
+Also added column `fk_currency` into  `spy_store` for save default currency per store.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/upgrade-modules/upgrade-the-currency-module.md:7]:
```suggestion
We also added the column `fk_currency` into  `spy_store`, to allow saving a default currency per store.
```

File: _includes/pbc/all/upgrade-modules/upgrade-the-currency-module.md [hunk 13]
```diff
@@ -1,4 +1,29 @@
...
+
+You can find more details about the changes on the [Currency module](https://github.com/spryker/currency/releases) release page.
+
+*Estimated migration time: 5 min*
```
Comment (AlexSlawinski) [inline _includes/pbc/all/upgrade-modules/upgrade-the-currency-module.md:11]:
```suggestion
*Estimated migration time: 5 minutes*
```

File: _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md [hunk 14]
```diff
@@ -1,3 +1,30 @@
+This document describes how to upgrade the Locale module.
+
+## Upgrading from version 3.* to version 4.0.0
+
+In this new version of the `Locale` module, we have added support configuration locale for each store in database.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md:5]:
```suggestion
In this new 4.0.0 version of the `Locale` module, we have added support configuration locale for each store in the database.
```

File: _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md [hunk 15]
```diff
@@ -1,3 +1,30 @@
...
+With the `Locale` module version 4 we have added the `spy_locale_store` database table to persist stores-locales in Zed.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md:6]:
```suggestion
With the new 4.0.0 version of the `Locale` module, we have also added the `spy_locale_store` database table to persist stores-locales in Zed.
```

File: _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md [hunk 16]
```diff
@@ -1,3 +1,30 @@
...
+Also added column `fk_locale` into  `spy_store` for save default locale per store.
```
Comment (AlexSlawinski) [inline _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md:7]:
```suggestion
We also added the column `fk_locale` into  `spy_store`, to allow saving a default locale per store. 
```

File: _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md [hunk 17]
```diff
@@ -1,3 +1,30 @@
...
+
+## Upgrading from version 3.* to version 4.0.0
+
+In this new version of the `Locale` module, we have added support configuration locale for each store in database.
+With the `Locale` module version 4 we have added the `spy_locale_store` database table to persist stores-locales in Zed.
+Also added column `fk_locale` into  `spy_store` for save default locale per store. 
+
+You can find more details about the changes on the [Locale module](https://github.com/spryker/locale/releases) release page.
+
+*Estimated migration time: 5 min*
```
Comment (AlexSlawinski) [inline _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md:11]:
```suggestion
*Estimated migration time: 5 minutes*
```

**Review by AlexSlawinski — APPROVED** [review]

**Pull request merged**